src/ucp/core/ucp_context.h

```diff
@@ -330,7 +336,8 @@ ucp_memory_type_detect_mds(ucp_context_h context, void *addr, size_t length,
     }
 
     if (context->memtype_cache != NULL) {
-        if (ucs_memtype_cache_lookup(context->memtype_cache, addr,
+        if (!ucp_memory_type_cache_is_empty(context) &&
```

Comment:
why need to check it's empty before lookup?

Reply:
@yosefe actually no need. but i thought it might reduce some instr cycles in pure host application case.  WDYT?

Reply:
you mean for non-inline messages, right? and for RX flow?

Reply:
yes.  ucp_memory_type_detect_mds is already called from  RX flow. and it there is no regression in pure host case with default cuda modules loading.